daemon/containerd/mount.go

```diff
@@ -38,12 +42,14 @@ func (i *ImageService) Mount(ctx context.Context, container *container.Container
 func (i *ImageService) Unmount(ctx context.Context, container *container.Container) error {
 	root := container.BaseFS
 
-	if err := mount.UnmountAll(root, 0); err != nil {
-		return fmt.Errorf("failed to unmount %s: %w", root, err)
-	}
+	if container.State != nil && !container.State.Running {
```

Comment:
Could it happen that these conditions were true for `Mount` but not for `Unmount` (state changed between these invocations)?
Maybe it would be better safer to check the actual `container.BaseFS` instead? For example:
```suggestion
    if strings.HasSuffix(root, "_rootfs-mount")
```